packages/react-instantsearch-core-next/src/useConnector.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+
+import { useIndexContext } from './useIndexContext';
+import { useStableValue } from './useStableValue';
+
+import type {
+  Connector,
+  WidgetDescription,
+  WidgetRenderState,
+} from 'instantsearch.js';
+
+type ConnectorState<TProps> = Omit<
+  WidgetRenderState<any, TProps>,
+  'widgetParams'
+>;
+
+export function useConnector<
+  TProps extends Record<string, unknown>,
+  TState extends ConnectorState<TProps>,
+  TDescription extends WidgetDescription
+>(
+  connector: Connector<TDescription, TProps>,
+  props: TProps = {} as TProps,
+  initialState: TState | (() => TState)
+) {
+  const parentIndex = useIndexContext();
+  const [state, setState] = useState(initialState);
+  const stableProps = useStableValue(props);
+
+  useEffect(() => {
+    const createWidget = connector((connectorState, isFirstRender) => {
+      // We skip the `init` widget render because:
+      // - We rely on a computed initial state before the InstantSearch Core lifecycle starts.
+      // - It prevents UI flashes when updating the widget props.
+      if (isFirstRender) {
+        return;
+      }
+
+      const { instantSearchInstance, widgetParams, ...renderState } =
+        connectorState;
+
+      setState(renderState as ConnectorState<TProps>);
```

Comment:
Why do you need to cast?

Reply:
No need anymore since 32c4ed8fcf12a54045043ab233eb849243659503 🙂